src/keyboard.js

```diff
@@ -0,0 +1,187 @@
+import buttonsMap from './buttonsmap.js';
+
+class Keyboard {
+  constructor() {
+    this.lang = localStorage.getItem('keyboardLang') || 'en';
+    this.capitalisation = 'normal';
+    this.capslocked = false;
+  }
+
+  addButtons() {
+    const fragment = document.createDocumentFragment();
+    const keyCodes = Object.keys(buttonsMap);
+    keyCodes.forEach((x) => {
+      const button = document.createElement('div');
+      button.textContent = buttonsMap[x].key[this.capitalisation][this.lang];
+      button.classList.add('keyboard__button');
+      button.classList.add(`keyboard__button_width_${buttonsMap[x].width}`);
+      button.dataset.code = x;
+      fragment.appendChild(button);
+    });
+
+    return fragment;
+  }
+
+  switchLanguage() {
+    this.lang = this.lang === 'en' ? 'ru' : 'en';
+  }
+
+  shiftCapitalisation() {
+    this.capitalisation = this.capitalisation === 'normal' ? 'shifted' : 'normal';
+  }
+
+  toggleCapslock() {
+    this.capslocked = this.capslocked !== true;
+  }
+
+  drawButtons() {
+    const keyboardButtons = document.querySelectorAll('.keyboard__button');
+    for (let i = 0; i < keyboardButtons.length; i += 1) {
+      keyboardButtons[i].textContent = buttonsMap[keyboardButtons[i].dataset.code]
+        .key[this.capitalisation][this.lang];
+    }
+  }
+
+  init() {
+    const wrapper = document.createElement('div');
+    wrapper.classList.add('wrapper');
+    const textarea = document.createElement('textarea');
+    textarea.classList.add('textarea');
+    wrapper.appendChild(textarea);
+    const keyboard = document.createElement('div');
+    keyboard.classList.add('keyboard');
+    const info = document.createElement('div');
+    info.innerHTML = '<div class="info__hint"><p>Switch language combo: <span>alt</span> + <span>shift</span></p></div>'
+    + '<div class="info__os"><p>Developed for WINDOWS OS<p></div>';
+    info.classList.add('info');
+    wrapper.appendChild(keyboard);
+    keyboard.appendChild(this.addButtons());
+    wrapper.appendChild(info);
+    document.body.appendChild(wrapper);
+    const shiftKeys = document.querySelectorAll('[data-code*="Shift');
+    const capslockKey = document.querySelector('[data-code="CapsLock"');
+    capslockKey.classList.add('keyboard__button_capslock');
+
+    document.addEventListener('keydown', (evt) => {
+      if (Object.keys(buttonsMap).includes(evt.code)) {
+        evt.preventDefault();
+        if ((evt.code === 'ShiftLeft' || evt.code === 'ShiftRight')) {
+          if (!Array.from(shiftKeys).some((x) => x.classList.contains('keyboard__button_active'))) {
```

Comment:
please replace X to more meaningful variable name

Reply:
done.